sql/src/main/java/io/crate/expression/scalar/arithmetic/ArithmeticFunctions.java

```diff
@@ -208,4 +279,140 @@ public static Function of(String name, Symbol first, Symbol second, Set<Function
             ImmutableList.of(first, second)
         );
     }
+
+    private static final class IntervalNumberScalar extends Scalar<Period, Object> {
+
+        private final FunctionInfo info;
+        private final BiFunction<BigDecimal, BigDecimal, BigDecimal> operation;
+
+        IntervalNumberScalar(String operator, DataType numericalType) {
+            this.info = new FunctionInfo(
+                new FunctionIdent("add", Arrays.asList(DataTypes.INTERVAL, numericalType)), DataTypes.INTERVAL);
+            switch (operator) {
+                case "/":
+                    this.operation = (a,b) -> a.divide(b, RoundingMode.HALF_EVEN);
+                    break;
+                case "*":
+                    this.operation = (a,b) -> a.multiply(b, MathContext.DECIMAL64);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unsupported operator for interval " + operator);
+            }
+        }
+
+        @Override
+        public Period evaluate(TransactionContext txnCtx, Input<Object>... args) {
+            Period period = (Period) args[0].value();
+            Object number = args[1].value();
+
+            BigDecimal millis = new BigDecimal(period.toStandardDuration().getMillis());
```

Comment:
This will fail if the `period` contains months. 

Could we maybe extract this logic here into a separate PR and wrap up the `timestamp +/- interval` cases first?

Reply:
Good idea.